cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/query/IndexUpdater.java

```diff
@@ -199,9 +199,6 @@ private void updateIndex(final String indexName,
                                                             initialArgs,
                                                             indexName,
                                                             rev);
-                if (parameter == null) {
```

Comment:
I can see this and it's friend a few lines further on were removed because `com.cloudant.sync.internal.query.IndexUpdater#populateDBParameter` doesn't appear to be able to return null. However, I wonder if that method actually needs looking at because there is no `else` case for if we couldn't figure out the type and then not indexing that field.